src/main/java/org/broadinstitute/hellbender/tools/LocalAssembler.java

```diff
@@ -0,0 +1,2306 @@
+                    case 'G': case 'g': kVal += 2; break;
+                    case 'T': case 't': kVal += 3; break;
+                }
+                if ( ++count >= Kmer.KSIZE ) {
+                    final KmerAdjacency kmer = KmerAdjacencyImpl.find(kVal, kmerAdjacencySet);
+                    // if we fail to look up the kmer
+                    if ( kmer == null ) {
+                        if ( currentPathPart == null ) {
+                            // if there's no current path part, just create the 1st one as a PathPartGap
+                            currentPathPart = new PathPartGap(new KmerAdjacencyImpl(kVal));
+                            parts.add(currentPathPart);
+                        } else if ( currentPathPart.isGap() ) {
+                            // if the current path part is a PathPartGap, just extend it
+                            currentPathPart.extend(call);
+                        } else {
+                            // new PathPartGap
+                            currentPathPart = new PathPartGap(new KmerAdjacencyImpl(kVal));
+                            parts.add(currentPathPart);
+                        }
+                    } else {
+                        // we've found our kmer
+                        final Contig contig = kmer.getContig();
+                        if ( currentPathPart == null ) {
+                            // we've looked up a kmer, but don't have a current path part -- create one
+                            currentPathPart = new PathPartContig(contig, kmer.getContigOffset());
+                            parts.add(currentPathPart);
+                        } else if ( contig == currentPathPart.getContig() ) {
+                            // our lookup is on the current path part's contig -- extend it
+                            final int kmerOffset = kmer.getContigOffset();
+                            final int curStop = currentPathPart.getStop();
+                            if ( kmerOffset == curStop ) {
+                                currentPathPart.extend(call);
+                            } else if ( kmerOffset == 0 && contig.getNKmers() == curStop ) {
+                                // cycle onto same contig
+                                currentPathPart = new PathPartContig(contig, 0);
+                                parts.add(currentPathPart);
+                            } else {
+                                // weird:  kmer is non-contiguous.  start a new path part after a zero-length gap
+                                parts.add(zeroLengthGap(currentPathPart));
+                                currentPathPart = new PathPartContig(contig, kmerOffset);
+                                parts.add(currentPathPart);
+                            }
+                        } else {
+                            final int kmerContigOffset = kmer.getContigOffset();
+                            if ( currentPathPart.isGap() ) {
```

Comment:
This is the transition out of a gap back to a contig, right? Otherwise we wouldn't have found our kmer in the set?